docs/en/stack/ml/anomaly-detection/anomaly-detection-scale.asciidoc

```diff
@@ -0,0 +1,228 @@
+[role="xpack"]
+[[anomaly-detection-scale]]
+= Using {anomaly-detect} at scale
+
+Working with large {anomaly-jobs} requires careful planning; otherwise, job 
+performance could be adversely effected. This page contains several guidelines 
+to consider when using {anomaly-detect} at scale and helps you to avoid possible 
+pitfalls.
+
+
+[[infrastructure-config]]
+== Infrastructure configuration guidelines
+
+
+[[number-ml-nodes]]
+=== Deciding the number of {ml} nodes 
+
+By using the 
+{ref}//ml-estimate-model-memory.html[Estimate {anomaly-jobs} model memory API], 
+you can get approximately how much memory each job will require. Based on this 
+information, decide how many {ml} nodes you need to add to the cluster to 
+support those jobs. A single job runs on a single node. If you have a job that 
+requires 10 GB of memory to run, you need a {ml} node that can provide enough 
+resources to run the job.
+
+You can control how much of the available native memory the node can use for 
+{ml} processes (these processes are separate to {es} JVM) via the 
+`xpack.ml.max_machine_memory_percent` setting. If the node is a dedicated {ml} 
+node, increase this value from the default 30%. **If you have a cloud 
+deployment, you don't need to manually increase this value, it is automatically 
+set.**
+
+
+[[max-model-memory-limit]]
+=== Setting model memory limit properly
+
+There are two model memory limit configuration options:
+
+* `xpack.ml.max_model_memory_limit` is a node setting that defines the maximum 
+  `model_memory_limit` value you can set for any {anomaly-job} on the node. If 
+  the memory requirement of a job is higher than this value, an error occurs 
+  and the job cannot be created.
+
+* `model_memory_limit` is a job variable that sets the approximate maximum 
```

Comment:
@sophiec20 Yes, you are right it is a job config section. However, memory settings have two aspects: one is infrastructural, the other is a job config. I think it is better to talk about them in one place rather than split the two aspects as 1. they are interconnected, 2. it would be harder to follow the story. What do you think?

Reply:
Some users have the privilege to set cluster/node settings, some only job settings. Mixing up node setting with job settings is probably more confusing from a users perspective. 

Plus I'm not sure how common it is for them to be connected. If the job is in HARD_LIMIT, it's generally because the job config was insufficient, not because this node setting was limiting it.. 

In cloud, we set this to the appropriate value for the selected node size. Thus stopping anyone from creating a job that would have no chance fitting on the node. We would not advise end-users to increase this value (even if they could). In the spirit of cloud-first documentation, I think this node setting is distracting.

When working at scale, you need to choose a model_memory_limit that is high enough for your job to work without hitting hard_limit, and this may mean scaling up your cluster because you won’t be allowed to use a high model_memory_limit in a small (Cloud) cluster. This point is already made in the prev section I think.

Reply:
@sophiec20 Okay, thanks for the explanation. I'll add `model_memory_limit` to the job config section and remove the `xpack.ml.max_model_memory_limit` reference from the docs as it is distracting.